common/app/common/JsonComponent.scala

```diff
@@ -12,6 +12,12 @@ import play.api.http.ContentTypes._
 
 object JsonComponent extends Results with implicits.Requests {
 
+  def withRefreshStatus(json: JsValue): JsValue =
+    json.as[JsObject] + ("refreshStatus" -> toJson(AutoRefreshSwitch.isSwitchedOn))
+
+  def apply[A](value: A)(implicit request: RequestHeader, writes:Writes[A]) =
+    resultFor(request, Json.stringify(withRefreshStatus(toJson(value))))
```

Comment:
The problem with calling `withRefreshStatus` (and therefore value.as[JsObject]) is that it can blow up at runtime because you can have a value from a type with an implicit `Writes` but that cannot be transformed into a `JsObject` (ex: a `JsString`)